presto-druid/src/main/java/io/prestosql/plugin/druid/DruidJdbcClient.java

```diff
@@ -267,4 +294,76 @@ public void dropSchema(JdbcIdentity identity, String schemaName)
+                .add(new ImplementCountAll(bigIntHandle))
+                .add(new ImplementCount(bigIntHandle))
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("approx_distinct")
+                        .druidName("approx_count_distinct")
+                        .jdbcTypeHandle(bigIntHandle)
+                        .outputType(BIGINT)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("max")
+                        .inputTypes(numericAndTimeStampType)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("min")
+                        .inputTypes(numericAndTimeStampType)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("sum")
+                        .inputTypes(druidNumericTypes)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("avg")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("stddev")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("stddev_pop")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("stddev_samp")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("variance")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("var_pop")
+                        .jdbcTypeHandle(doubleHandle)
+                        .inputTypes(druidNumericTypes)
+                        .outputType(DOUBLE)
+                        .build())
+                .add(SingleInputAggregateFunction.builder()
+                        .prestoName("var_samp")
```

Comment:
The empty and single-row cases does not return null in druid's case like presto. I think the larger question we need to answer is when a user is querying through presto what is the behavior they should expect. If we agree that they should expect presto's behavior then we have to either remove pushdowns for all aggregate functions that does not have same behavior or introduce another config so users can specify that they actually want the connector's behavior and not presto behavior. Based on what we decide I can make changes and post updated the PR. I have addressed rest of the comments and made changes to add a source with single row and no data at all but commented out the tests that would fail.

Reply:
General rule is -- correctness first.
I do no feel comfortable deviating from the rule without having a broader conversation with @martint and @electrum.

However, you perhaps are able to get Presto semantics out of Druid with `case count(*) when 0 then ... when 1 then ...`?

btw @martint do we have `var_samp` semantics spec-wise?

Reply:
Instead of waiting longer on this I have just modified the PR so the connector will just return what presto would have returned without pushdown. If our users need the non presto behavior, we will do something internally to support that.